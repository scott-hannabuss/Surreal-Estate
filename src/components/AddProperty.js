import React, { useState } from "react";
import Button from "react-uikit-button";
import axios from "axios";
import Alert from "./Alert";
import "../styles/AddProperty.css";

const AddProperty = () => {
  const initialState = {
    fields: {
      title: "",
      city: "Manchester",
      type: "",
      bathrooms: "",
      bedrooms: "",
      email: "",
      price: "",
    },
    alert: {
      message: "",
      isSuccess: false,
    },
  };

  const [fields, setFields] = useState(initialState.fields);
  const [alert, setAlert] = useState(initialState.alert);

  const handleAddProperty = (e) => {
    e.preventDefault();
    setAlert({ message: "", isSuccess: false });
    axios
      .post(
        `https://surrealestatedatabase.herokuapp.com/PropertyListing`,
        fields
      )
      .then(() =>
        setAlert({
          message: "",
          isSuccess: true,
        })
      )
      .catch(() =>
        setAlert({
          message: "",
          isSuccess: false,
        })
      );
  };

  const handleFieldChange = (event) => {
    setFields({ ...fields, [event.target.name]: event.target.value });
  };

  return (
    <div className="addproperty">
      <h1 className="header_addproperty">Add Property</h1>
      <Alert message={alert.message} success={alert.isSuccess} />
      <form className="add-property_form" onSubmit={handleAddProperty}>
        <label htmlFor="title">
          <input
            placeholder="Property Title"
            id="title"
            name="title"
            value={fields.title}
            onChange={handleFieldChange}
          />
        </label>
        <label htmlFor="city">
          <select
            placeholder="city"
            id="city"
            name="city"
            value={fields.city}
            onChange={handleFieldChange}
          >
            <option value="Manchester">Manchester</option>
            <option value="Leeds">Leeds</option>
            <option value="Sheffield">Sheffield</option>
            <option value="Liverpool">Liverpool</option>
          </select>
        </label>
        <label htmlFor="type">
          <select
            placeholder="Property Type"
            id="type"
            name="type"
            value={fields.type}
            onChange={handleFieldChange}
          >
            <option value="Flat">Flat</option>
            <option value="Detached">Detached</option>
            <option value="Semi-Detached">Semi-Detached</option>
            <option value="Terraced">Terraced</option>
            <option value="End Terraced">End Terraced</option>
            <option value="Cottage">Cottage</option>
            <option value="Bungalow">Bungalow</option>
          </select>
        </label>
        <label data-testid="test-bathroom-label" htmlFor="bathroom">
          <select
            placeholder="Number of bathrooms"
            id="bathroom"
            name="bathroom"
            value={fields.bathroom}
            onChange={handleFieldChange}
          >
            <option value="1 Bathroom">1 Bathroom</option>
            <option value="2 Bathroom">2 Bathroom</option>
            <option value="3 Bathroom">3 Bathroom</option>
          </select>
        </label>
        <label data-testid="test-bedroom-label" htmlFor="bedrooms">
          <select
            placeholder="Number of bedrooms"
            id="bedrooms"
            name="bedrooms"
            value={fields.bedrooms}
            onChange={handleFieldChange}
          >
            <option value="1 Bedrooms">1 Bedroom</option>
            <option value="2 Bedrooms">2 Bedrooms</option>
            <option value="3 Bedrooms">3 Bedrooms</option>
            <option value="4 Bedrooms">4 Bedrooms</option>
            <option value="5 Bedrooms">5 Bedrooms</option>
          </select>
        </label>
        <label htmlFor="email">
          <input
            placeholder="Email"
            id="email"
            name="email"
            value={fields.email}
            onChange={handleFieldChange}
          />
        </label>
        <label htmlFor="price">
          <input
            placeholder="Price"
            id="price"
            name="price"
            type="text"
            inputMode="numeric"
            pattern="[0-9]+"
            value={fields.price}
            onChange={handleFieldChange}
          />
        </label>
        <Button className="add-property-button" type="submit">
          Add Property!
        </Button>
      </form>
    </div>
  );
};

export default AddProperty;
