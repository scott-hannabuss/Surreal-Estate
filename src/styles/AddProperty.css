@import url("https://fonts.googleapis.com/css2?family=Arvo:wght@400;700&family=Cairo:wght@700&family=Dela+Gothic+One&family=Lato&family=Noto+Sans+TC:wght@700&display=swap");

.addproperty {
  text-align: center;
  padding: 10px;
  background: url(../images/mansion.jpeg);
  background-repeat: no-repeat;
  background-attachment: fixed;
  background-position: center;
  background-size: cover;
  min-width: 100%;
  min-height: 100%;
}

.header_addproperty {
  color: white;
  font-family: "Lato", sans-serif;
  margin-top: 2%;
}

.add-property_form {
  border: 1px solid white;
  text-align: center;
  padding: 5%;
  margin-top: 1%;
  margin-bottom: 5%;
  width: 50%;
  padding: 1em;
  background-color: rgba(255, 255, 255, 0.9);
  box-shadow: 0 8px 32px 0 rgba(31, 38, 135, 0.6);
  -webkit-backdrop-filter: blur(15.5px);
  border-radius: 1em;
  color: black;
  font-size: 3vmin;
  font-family: "Lato", sans-serif;
  display: inline-block;
}

.add-property_form label {
  display: block;
  font-family: "Lato", sans-serif;
  width: 100%;
  margin: auto;
  margin-bottom: 2%;
  text-align: center;
  border-radius: 25px;
}

.add-property_form input {
  width: 48%;
  margin-bottom: 2%;
  text-align: center;
  font-family: "Lato", sans-serif;
  border-radius: 25px;
}

.add-property_form select {
  width: 50%;
  margin-bottom: 2%;
  text-align: center;
  font-family: "Lato", sans-serif;
  border-radius: 25px;
}

.add-property-button {
  font-family: "Lato", sans-serif;
  padding: 1%;
  border-radius: 25px;
  width: 40%;
  background-color: #12093b;
  color: white;
  cursor: pointer;
}

@media only screen and (min-width: 751px) and (max-width: 1025px) {
  .add-property_form {
    height: 100vw;
  }
  .header_addproperty {
    margin-top: 3%;
    margin-bottom: 3%;
    font-size: 3em;
  }
  .add-property_form input {
    width: 70%;
    height: 2em;
    margin-bottom: 2%;
    margin-top: 2%;
    font-size: 0.7em;
  }
  .add-property_form select {
    margin-bottom: 2%;
    margin-top: 2%;
    width: 70%;
    height: 2em;
    font-size: 0.7em;
    font-family: "Lato", sans-serif;
    text-align: center;
  }
  .add-property-button {
    margin-top: 2%;
    margin-bottom: 1%;
    width: 60%;
    height: 2em;
    font-size: 0.9em;
  }
}
