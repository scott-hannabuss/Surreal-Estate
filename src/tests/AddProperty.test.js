import React from "react";
import { render } from "@testing-library/react";
import AddProperty from "../components/AddProperty";

describe("AddProperty", () => {
  it("renders as expected", () => {
    const { asFragment } = render(<AddProperty />);
    expect(asFragment()).toMatchSnapshot();
  });
});

it("renders a bathroom label", () => {
  const { getByTestId } = render(<AddProperty />);
  const bathroomLabel = getByTestId("test-bathroom-label");
  expect(bathroomLabel).toBeTruthy();
});

it("renders a bedroom label", () => {
  const { getByTestId } = render(<AddProperty />);
  const bedroomLabel = getByTestId("test-bedroom-label");
  expect(bedroomLabel).toBeTruthy();
});
