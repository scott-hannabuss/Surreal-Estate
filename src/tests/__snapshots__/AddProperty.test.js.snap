// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProperty renders as expected 1`] = `
<DocumentFragment>
  <div
    class="addproperty"
  >
    <h1>
      Add Property
    </h1>
    <form>
      <label
        for="title"
      >
        <input
          id="title"
          name="title"
          placeholder="title"
          value=""
        />
      </label>
      <label
        for="city"
      >
        <select
          id="city"
          name="city"
          placeholder="city"
        >
          <option
            value="Manchester"
          >
            Manchester
          </option>
          <option
            value="Leeds"
          >
            Leeds
          </option>
          <option
            value="Sheffield"
          >
            Sheffield
          </option>
          <option
            value="Liverpool"
          >
            Liverpool
          </option>
        </select>
      </label>
      <label
        for="type"
      >
        <select
          id="type"
          name="type"
          placeholder="Property Type"
        >
          <option
            value="Flat"
          >
            Flat
          </option>
          <option
            value="Detached"
          >
            Detached
          </option>
          <option
            value="Semi-Detached"
          >
            Semi-Detached
          </option>
          <option
            value="Terraced"
          >
            Terraced
          </option>
          <option
            value="End Terraced"
          >
            End Terraced
          </option>
          <option
            value="Cottage"
          >
            Cottage
          </option>
          <option
            value="Bungalow"
          >
            Bungalow
          </option>
        </select>
      </label>
      <label
        data-testid="test-bathroom-label"
        for="bathroom"
      >
        <select
          id="bathroom"
          name="bathroom"
          placeholder="Number of bathrooms"
        >
          <option
            value="1 Bathroom"
          >
            1 Bathroom
          </option>
          <option
            value="2 Bathroom"
          >
            2 Bathroom
          </option>
          <option
            value="3 Bathroom"
          >
            3 Bathroom
          </option>
        </select>
      </label>
      <label
        data-testid="test-bedroom-label"
        for="bedrooms"
      >
        <select
          id="bedrooms"
          name="bedrooms"
          placeholder="Number of bedrooms"
        >
          <option
            value="1 Bedrooms"
          >
            1 Bedroom
          </option>
          <option
            value="2 Bedrooms"
          >
            2 Bedrooms
          </option>
          <option
            value="3 Bedrooms"
          >
            3 Bedrooms
          </option>
          <option
            value="4 Bedrooms"
          >
            4 Bedrooms
          </option>
          <option
            value="5 Bedrooms"
          >
            5 Bedrooms
          </option>
        </select>
      </label>
      <label
        for="email"
      >
        <input
          id="email"
          name="email"
          placeholder="email"
          value=""
        />
      </label>
      <label
        for="price"
      >
        <input
          id="price"
          inputmode="numeric"
          name="price"
          pattern="[0-9]+"
          placeholder="price"
          type="text"
          value=""
        />
      </label>
      <button
        class="uk-button"
        data-kitid="ckmxayhb700009qtlk8j2q7mi"
        type="submit"
      >
        Add
      </button>
    </form>
  </div>
</DocumentFragment>
`;
